regression/cbmc/symex_should_evaluate_simple_pointer_conditions/test.c

```diff
@@ -0,0 +1,76 @@
+#include <assert.h>
+
+static void noop() { }
+
+int test(int nondet_int) {
+  int a = 1, b = 2, c = 3;
+
+  // Symex knows the value of ptr_to_a, so it should be able to evaluate simple
+  // conditions involving it.
+  int *ptr_to_a = &a;
+
+  int unconditionally_reachable_1;
+  if(ptr_to_a == &a)
+    unconditionally_reachable_1 = 7;
+
+  int unreachable_2;
+  if(ptr_to_a == &c)
+    unreachable_2 = 7;
+
+  int unconditionally_reachable_3;
+  if(ptr_to_a != &c)
+    unconditionally_reachable_3 = 7;
+
+  int unreachable_4;
+  if(ptr_to_a != &a)
+    unreachable_4 = 7;
+
+  // Symex can't tell what ptr_to_a_or_b points to, but we can tell that it
+  // doesn't point to some things
+  int *ptr_to_a_or_b = nondet_int == 0 ? &a : &b;
+
+  int possibly_reachable_5;
+  if(ptr_to_a_or_b == &a)
+    possibly_reachable_5 = 7;
+
+  int unreachable_6;
+  if(ptr_to_a_or_b == &c)
+    unreachable_6 = 7;
+
+  int unconditionally_reachable_7;
+  if(ptr_to_a_or_b != &c)
+    unconditionally_reachable_7 = 7;
+
+  int possibly_reachable_8;
+  if(ptr_to_a_or_b != &a)
+    possibly_reachable_8 = 7;
+
+  // We should also be able to do all of the above in compound expressions which
+  // use logical operators like AND, OR and NOT.
+
+  int unconditionally_reachable_9;
+  if(!(ptr_to_a == &c))
+    unconditionally_reachable_9 = 7;
+
+  int unreachable_10;
+  if(!(ptr_to_a == &a))
+    unreachable_10 = 7;
+
+  int unconditionally_reachable_11;
+  if(!(ptr_to_a_or_b == &c))
+    unconditionally_reachable_11 = 7;
+
+  int possibly_reachable_12;
+  if(!(ptr_to_a_or_b == &a))
+    possibly_reachable_12 = 7;
+
+  int unconditionally_reachable_13;
+  if((ptr_to_a == &a && c == 3) || c == 0)
```

Comment:
Bear in mind these constant comparisons with `c` may well be resolved very early as `c` is only assigned once and never address-taken.

Reply:
I've changed them to other pointer comparisons. And I now take the address of all the integers at the beginning, if that's a thing that symex cares about.